sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -820,12 +820,27 @@ def request_status(self, status_link: str) -> PipelineResponse[HttpRequestTypeVa
         """
         if self._path_format_arguments:
             status_link = self._client.format_url(status_link, **self._path_format_arguments)
+        request_params: Dict[str, Any] = {}
+        if self._lro_options and "api_version" in self._lro_options:
+            parsed_status_link = urllib.parse.urlparse(status_link)
+            request_params = cast(
+                Dict[str, Any],
+                case_insensitive_dict(
```

Comment:
You can use a regular dict, there is no reason to use `case_insensitive_dict` and a cast here